
--- SRC.orig/dlarft.f	2009-04-16 21:09:59.000000000 +0200
+++ SRC/dlarft.f	2009-04-24 15:59:38.000000000 +0200
@@ -144,10 +144,11 @@
                VII = V( I, I )
                V( I, I ) = ONE
                IF( LSAME( STOREV, 'C' ) ) THEN
-!                 Skip any trailing zeros.
+*                 Skip any trailing zeros.
                   DO LASTV = N, I+1, -1
-                     IF( V( LASTV, I ).NE.ZERO ) EXIT
+                     IF( V( LASTV, I ).NE.ZERO ) GOTO 666
                   END DO
+ 666              CONTINUE
                   J = MIN( LASTV, PREVLASTV )
 *
 *                 T(1:i-1,i) := - tau(i) * V(i:j,1:i-1)' * V(i:j,i)
@@ -156,10 +157,11 @@
      $                        V( I, 1 ), LDV, V( I, I ), 1, ZERO,
      $                        T( 1, I ), 1 )
                ELSE
-!                 Skip any trailing zeros.
+*                 Skip any trailing zeros.
                   DO LASTV = N, I+1, -1
-                     IF( V( I, LASTV ).NE.ZERO ) EXIT
+                     IF( V( I, LASTV ).NE.ZERO ) GOTO 667
                   END DO
+ 667              CONTINUE
                   J = MIN( LASTV, PREVLASTV )
 *
 *                 T(1:i-1,i) := - tau(i) * V(1:i-1,i:j) * V(i,i:j)'
@@ -200,10 +202,11 @@
                   IF( LSAME( STOREV, 'C' ) ) THEN
                      VII = V( N-K+I, I )
                      V( N-K+I, I ) = ONE
-!                    Skip any leading zeros.
+*                    Skip any leading zeros.
                      DO LASTV = 1, I-1
-                        IF( V( LASTV, I ).NE.ZERO ) EXIT
+                        IF( V( LASTV, I ).NE.ZERO ) GOTO 668
                      END DO
+ 668                 CONTINUE
                      J = MAX( LASTV, PREVLASTV )
 *
 *                    T(i+1:k,i) :=
@@ -216,10 +219,11 @@
                   ELSE
                      VII = V( I, N-K+I )
                      V( I, N-K+I ) = ONE
-!                    Skip any leading zeros.
+*                    Skip any leading zeros.
                      DO LASTV = 1, I-1
-                        IF( V( I, LASTV ).NE.ZERO ) EXIT
+                        IF( V( I, LASTV ).NE.ZERO ) GOTO 669
                      END DO
+ 669                 CONTINUE
                      J = MAX( LASTV, PREVLASTV )
 *
 *                    T(i+1:k,i) :=
